lib/base.js

```diff
@@ -1,138 +1,69 @@
-      // to do with the `content`, but downstream software
-      // may know.
-      return BaseElement;
-    }
-
-    return ElementClass;
-  },
-
-  /*
-   * Convert a long-form refract document into refract element instances.
-   */
-   fromRefract: function(doc) {
-    var ElementClass = this.getElementClass(doc.element);
-    return new ElementClass().fromRefract(doc);
-  },
-
-  /*
-   * Convert a compact refract tuple into refract element instances.
-   */
-  fromCompactRefract: function(tuple) {
-    var ElementClass = this.getElementClass(tuple[0]);
-    return new ElementClass().fromCompactRefract(tuple);
+exports.init = function(registry) {
+  // Allows for passing in a custom registry
+  if (!registry) {
+    registry = new ElementRegistry();
   }
-});
-
-// Initiate a default Minim registry
-var registry = new ElementRegistry();
 
-// Use dependency injection so we can split up into multiple files yet resolve
-// any circuluar dependencies. BaseElement is also defined above in order to
-// satisfy any linting issues.
-BaseElement = require('./primitives/base-element')(registry);
-var NullElement = require('./primitives/null-element')(BaseElement);
-var StringElement = require('./primitives/string-element')(BaseElement);
-var NumberElement = require('./primitives/number-element')(BaseElement);
-var BooleanElement = require('./primitives/boolean-element')(BaseElement);
-var ArrayElement = require('./primitives/array-element')(BaseElement, registry);
-var MemberElement = require('./primitives/member-element')(BaseElement, registry);
-var ObjectElement = require('./primitives/object-element')(BaseElement, ArrayElement, MemberElement);
-
-module.exports = {
-  BaseElement: BaseElement,
-  NullElement: NullElement,
-  StringElement: StringElement,
-  NumberElement: NumberElement,
-  BooleanElement: BooleanElement,
-  ArrayElement: ArrayElement,
-  MemberElement: MemberElement,
-  ObjectElement: ObjectElement,
-  ElementRegistry: ElementRegistry,
-  registry: registry
+  // Use dependency injection so we can split up into multiple files yet resolve
+  // any circuluar dependencies. BaseElement is also defined above in order to
+  // satisfy any linting issues.
+  var BaseElement = require('./primitives/base-element')(registry);
+
+  // The registry needs a base element to determine the base type of an element.
+  registry.BaseElement = BaseElement;
+
+  var NullElement = require('./primitives/null-element')(BaseElement);
+  var StringElement = require('./primitives/string-element')(BaseElement);
+  var NumberElement = require('./primitives/number-element')(BaseElement);
+  var BooleanElement = require('./primitives/boolean-element')(BaseElement);
+  var ArrayElement = require('./primitives/array-element')(BaseElement, registry);
+  var MemberElement = require('./primitives/member-element')(BaseElement, registry);
+  var ObjectElement = require('./primitives/object-element')(BaseElement, ArrayElement, MemberElement);
```

Comment:
This would be made possible per what you've suggested above.

https://github.com/refractproject/minim/pull/73#discussion_r37405835